### PR TITLE
Task 4: Integration with DynamoDB

### PR DESCRIPTION
**Pull request by glhvta**
**Task:**  
https://github.com/EPAM-JS-Competency-center/cloud-development-course-initial/blob/main/4_integration_with_database/task.md 

**PR Scope:**

- [x] Implemented DynamoDb integration using IaSS approach and serverless.yaml configuration
- [x] Integrated existing labmdas (`getProductsList` and `getProductById `) with DynamoDB
- [x] Created a new lambda function called `createProduct` under existing Serverless config file (i.e. serverless.yaml) of Product Service which is triggered by the HTTP POST method.
- [x] Added serverless-offline plugin and `launch.json` file for local lambda debugging
- [x] Added unit tests for `createProduct` lambda

**Additional (optional) tasks**
- [x] POST /products lambda functions returns error 400 status code if product data is invalid
- [x] All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
- [x] All lambdas do console.log for each incoming requests and their arguments

**Frontend application:**
https://d3jv20c7ho3whs.cloudfront.net/

**Endpoints:**
  GET - https://bta8xv58vl.execute-api.eu-central-1.amazonaws.com/dev/products
  GET - https://bta8xv58vl.execute-api.eu-central-1.amazonaws.com/dev/products/{productId}
  POST - https://bta8xv58vl.execute-api.eu-central-1.amazonaws.com/dev/products

### CODE REVIEW

File: product-service/src/functions/createProduct.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { APIGatewayEvent, APIGatewayProxyResult } from 'aws-lambda';
+import Ajv from 'ajv';
+
+import { ProductService } from '@libs/services/productsService';
+import { badRequestErrorResponse, serverErrorResponse, successResponse } from '@libs/services/responseBuilder';
+import { ProductRequestSchema, ProductRequest } from '../dto/product';
+
+const validateProduct = (new Ajv()).compile(ProductRequestSchema);
```
Comment (bogdan-panteleev) [inline product-service/src/functions/createProduct.ts:8]: you may use '@middy/validator'. it allows to add validation as a middleware

Comment (bogdan-panteleev) [reply product-service/src/functions/createProduct.ts:8]: https://middy.js.org/docs/middlewares/validator

File: product-service/src/functions/createProduct.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    let productRequest: ProductRequest;
+
+    try {
+      productRequest = JSON.parse(event.body) as ProductRequest;
+      
+      if (!validateProduct(productRequest)) {
+        throw new Error();
+      }
+    } catch(err) {
+      return badRequestErrorResponse('Invalid product data');
```
Comment (bogdan-panteleev) [inline product-service/src/functions/createProduct.ts:23]: middy also has some middlewares which may be suitable here

**Comment by glhvta** [comment]
@bogdan-panteleev, thanks a lot for the review and suggestions!

**Pull request merged**